app/controllers/reviews_controller.rb

```diff
@@ -0,0 +1,40 @@
+class ReviewsController < SecuredController
+  skip_before_action :authorize_request, only: %i[index show]
+
+  # GET /reviews
+  def index
+    user_sub = request.headers['body']
+    fab = User.find_by(user_sub: user_sub)
+    review = fab.reviews
+    render json: review
+  end
+
+  # GET /reviews/1
+  def show
+    review = Review.where(laptop_id: params[:id])
+    render json: review
+  end
+
+  # POST /reviews
+  def create
+    review = Review.create!(review_params)
+
+    render json: review, status: :created
+  end
+
+  # DELETE /reviews/1
+  def destroy
+    review = Review.find(params[:id])
+    review.delete
+    render json: review
+  end
+
+  private
+
+  def review_params
```

Comment:
Optional (but very important)

Our review_params method seems to be doing too much. Normally, we create this method to just sanitize parameters. But in this case we are also making it set a fab (by the way, fab is still using an unsanitized params "params[:user_sub]" which still leaves the app vulnerable to SQLi)

Also we're making it return something. Thus, when we run the method it creates an unnecessary object in memory.